server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -266,27 +242,6 @@ static void addCollectorsAndSearch(SearchContext searchContext) throws QueryPhas
         }
     }
 
-    private static void searchWithCollectorManager(
-        SearchContext searchContext,
-        ContextIndexSearcher searcher,
-        Query query,
-        CollectorManager<? extends Collector, Void> collectorManager,
-        boolean timeoutSet
-    ) throws IOException {
-        QuerySearchResult queryResult = searchContext.queryResult();
-        searcher.search(query, collectorManager);
-        if (searcher.timeExceeded()) {
-            assert timeoutSet : "TimeExceededException thrown even though timeout wasn't set";
-            if (searchContext.request().allowPartialSearchResults() == false) {
-                throw new QueryPhaseExecutionException(searchContext.shardTarget(), "Time exceeded");
-            }
-            queryResult.searchTimedOut(true);
-        }
-        if (searchContext.terminateAfter() != SearchContext.DEFAULT_TERMINATE_AFTER && queryResult.terminatedEarly() == null) {
-            queryResult.terminatedEarly(false);
-        }
-    }
```

Comment:
rather than having this method return a QueryPhaseResult, I folded it into its caller method, as that was the only caller for it.